tests/domains_test.go

```diff
@@ -1,21 +1,134 @@
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).Should(Say("%s", testApp.Name))
+			Eventually(sess).Should(Exit(0))
+
+			sess, err = start("deis domains:add %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("Adding %s to %s...", domain, testApp.Name))
+			Eventually(sess).Should(Say("done"))
+			Eventually(sess).Should(Exit(0))
+
+			sess, err = start("deis domains:list --app=%s", testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).Should(Say("%s", domain))
+			Eventually(sess).Should(Exit(0))
+
+			// curl app at both root and custom domain, both should return 200
+			curlCmd = Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, testApp.URL)}
+			Eventually(cmdWithRetry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
+			curlCmd = Cmd{CommandLineString: fmt.Sprintf(`curl -sL -H "Host: %s" -w "%%{http_code}\\n" "%s" -o /dev/null`, domain, testApp.URL)}
+			Eventually(cmdWithRetry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
+
+			sess, err = start("deis domains:remove %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("Removing %s from %s...", domain, testApp.Name))
+			Eventually(sess).Should(Say("done"))
+			Eventually(sess).Should(Exit(0))
+
+			// attempt to remove non-existent domain
+			sess, err = start("deis domains:remove %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("500 Internal Server Error")) // better error/explanation needed from cli
+			Eventually(sess).Should(Exit(1))
+
+			sess, err = start("deis domains:list --app=%s", testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).Should(Say("%s", testApp.Name))
+			Eventually(sess).Should(Not(Say("%s", domain)))
+			Eventually(sess).Should(Exit(0))
 
-		XIt("can add, list, and remove domains", func() {
-			// "deis domains:list --app=%s", app
-			// "deis domains:add %s --app=%s", domain, app
-			// "deis domains:list --app=%s", app
-			// curl app at both root and custom domain
-			// "deis domains:remove %s --app=%s", domain, app
-			// "deis domains:list --app=%s", app
 			// curl app at both root and custom domain, custom should fail
+			curlCmd = Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, testApp.URL)}
+			Eventually(cmdWithRetry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
+			curlCmd = Cmd{CommandLineString: fmt.Sprintf(`curl -sL -H "Host: %s" -w "%%{http_code}\\n" "%s" -o /dev/null`, domain, testApp.URL)}
+			Eventually(cmdWithRetry(curlCmd, strconv.Itoa(404), cmdRetryTimeout)).Should(BeTrue())
 		})
 
 		XIt("can add, list, and remove certs", func() {
```

Comment:
That's the one! Makes it easier to reason with since Certificates are considered a top level resources